### ISSUE TITLE
Upgrade to mio 0.4.x

### ISSUE THREAD

**Issue by zslayton** [zslayton/stomp-rs#94]
Currently running 0.3.x, which generates warnings when building. It will cease to work with rustc 1.3.

**Comment by frederikbosch** [comment]
We could even upgrade to mio 0.5.0 by now. This might also give this library support for windows. Nonetheless, I would love to see the upgrade. Do you see this happening any time soon? I will have look too!

**Comment by zslayton** [comment]
Agreed, targeting 0.5.x would be ideal; I'd love to be able to claim Windows support. I fully intend to do this but recently moved and started a new job, so I haven't had much time to work on my Github projects. 

I had hoped to migrate to the latest version of  `mio` by adapting `stomp-rs` to use [`mai`](https://github.com/zslayton/mai) under the covers. `mai` is another library I'm working on that exposes a high-level interface to `mio`'s functionality with the goal of making it easier to implement protocols asynchronously. When I last left off, `mai` had gotten to a point where I could start to modify `stomp-rs` but I never found the time. Hopefully I can pick this up again in the next couple of weeks!

**Comment by frederikbosch** [comment]
@zslayton That would be really awesome. I think it would be beneficial for both libraries - mai and stomp-rs - to find developers that can help us.

**Comment by zslayton** [comment]
I haven't forgotten about this. I'm going to work on it this week.

**Comment by zslayton** [comment]
This is proving to be quite the overhaul; the user-facing API is going to end up needing to change a fair amount. I should have an initial version by next weekend.

**Comment by frederikbosch** [comment]
@zslayton I can imagine. BC breaks for sure. But since we are below `1.0.0` this should be no problem.

**Comment by zslayton** [comment]
My work on this is now available in [this branch](https://github.com/zslayton/stomp-rs/tree/use-mai). It's working fine but there's a lot of cleanup to be done before I can promote it to master. The documentation in the README hasn't been updated yet. You can look at the `[send_and_subscribe](https://github.com/zslayton/stomp-rs/tree/use-mai)` example for a sense of what it'll look like when it's done.

**Comment by frederikbosch** [comment]
@zslayton Wonderful! I am going to test soon. You might want to create a PR at the mio package README and put `mia` in the [list of libraries that use mio](https://github.com/carllerche/mio#libraries). Maybe that helps getting people involved.
